server/cluster/cluster.go

```diff
@@ -703,14 +705,14 @@ func (c *RaftCluster) processRegionHeartbeat(region *core.RegionInfo) error {
 		c.prepareChecker.collect(region)
 	}
 
-	if c.regionStats != nil {
-		c.regionStats.Observe(region, c.getRegionStoresLocked(region))
-	}
-
 	changedRegions := c.changedRegions
-
 	c.Unlock()
 
+	for _, item := range overlaps {
+		statsJobs = append([]func(){func() { c.clearStats(item) }}, statsJobs...)
+	}
+	c.statsJobCh <- statsJobs
```

Comment:
If the channel is full. It can affect handling heartbeat.

Reply:
good idea